### PR TITLE
test: add test coverage for shared error utilities [DET-7900]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

Migrate error tests into shared and add more coverage.

## Test Plan

- [ ] ensure the tests pass locally
- [ ] ensure the tests pass via CI

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/shared/utils/error.test.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      expect(isDetError(detError)).toBe(true);
+      expect(detError.message).toBe(DEFAULT_ERROR_MESSAGE);
+    });
+
+    it('should construct DetError from Error', () => {
+      const error = new Error(DEFAULT_ERROR_MESSAGE);
+      const detError = new DetError(error);
+
+      expect(isError(error)).toBe(true);
+      expect(isDetError(error)).toBe(false);
```
Comment (johnkim-det) [inline webui/react/src/shared/utils/error.test.ts:72]: nit: are these two expects necessary?

File: webui/react/src/shared/utils/error.test.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  describe('isDetError', () => {
+    it('should detect DetError objects', () => {
+      const detError = new DetError('This is a crazy error!');
+      expect(isDetError(detError)).toBe(true);
+    });
+
+    it('should not report non-DetError items', () => {
+      const error = new Error('This is a normal error.');
+      expect(isDetError(error)).toBe(false);
```
Comment (johnkim-det) [inline webui/react/src/shared/utils/error.test.ts:51]: nit: might as well add boolean type like in the `isError` test?

File: webui/react/src/shared/utils/error.test.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      expect(detError.message).toBe(DEFAULT_ERROR_MESSAGE);
+    });
+
+    it('should construct DetError from DetError', () => {
+      const oldDetError = new DetError(DEFAULT_ERROR_MESSAGE);
+      const newDetError = new DetError(oldDetError);
+
+      expect(isError(oldDetError)).toBe(true);
+      expect(isDetError(oldDetError)).toBe(true);
+      expect(oldDetError.message).toBe(DEFAULT_ERROR_MESSAGE);
```
Comment (johnkim-det) [inline webui/react/src/shared/utils/error.test.ts:84]: ditto

File: webui/react/src/shared/utils/error.test.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    it('should construct DetError with error options', () => {
+      const oldDetError = new DetError(DEFAULT_ERROR_MESSAGE);
+      const newDetError = new DetError(oldDetError, DEFAULT_DET_ERROR_OPTIONS);
+
+      expect(isDetError(newDetError)).toBe(true);
+
+      // Expect each error option value to be preserved in the new DetError.
+      for (const [ key, value ] of Object.entries(DEFAULT_DET_ERROR_OPTIONS)) {
+        expect(value).toStrictEqual(DEFAULT_DET_ERROR_OPTIONS[key as keyof DetErrorOptions]);
```
Comment (johnkim-det) [inline webui/react/src/shared/utils/error.test.ts:98]: not sure if i understand this part. is this supposed to have `DEFAULT_DET_ERROR_OPTIONS` on both lines?

**Review by johnkim-det — COMMENTED** [review]
looks good, minor nits and one part i didn't understand

File: webui/react/src/shared/utils/error.test.ts (hunk 4 above)
Comment (hkang1) [reply webui/react/src/shared/utils/error.test.ts:98]: lol, good catch, supposed to have `newDetError` on one side

File: webui/react/src/shared/utils/error.test.ts (hunk 1 above)
Comment (hkang1) [reply webui/react/src/shared/utils/error.test.ts:72]: eh probably not, but just ensuring the error is an Error and not already a DetError...

**Review by johnkim-det — APPROVED** [review]
lgtm

**Pull request merged**